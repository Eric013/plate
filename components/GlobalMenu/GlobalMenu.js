import { inject, observer } from 'mobx-react'

import { Dimmer } from 'semantic-ui-react'
import Link from 'next/prefetch'
import Router from 'next/router'
import { auth } from '../../utils/db'

const logout = (closeMenu) => {
  auth.signOut()
  closeMenu()
  Router.push('/')
}

export default inject('GlobalStore')(observer(({ GlobalStore }) => (
  <div>
    <Dimmer active={GlobalStore.menuOpen}
      onClickOutside={GlobalStore.closeMenu}
      page
    >
      <div className='fade fadeIn'>
        <div className='menu-items'>
          <h1 onClick={GlobalStore.closeMenu}>
            <Link href='/'>
              <a>
                Home
              </a>
            </Link>
          </h1>
          {!GlobalStore.loggedIn &&
            <h1 onClick={GlobalStore.closeMenu}>
              <Link href='/login'>
                <a>
                  Login
                </a>
              </Link>
            </h1>
          }
          {GlobalStore.loggedIn &&
            <h1 onClick={() => logout(GlobalStore.closeMenu)}>
              <a>Logout</a>
            </h1>
          }
          {GlobalStore.loggedIn &&
            <h1 onClick={GlobalStore.closeMenu}>
              <Link href='/dashboard'>
                <a>
                  Dashboard
                </a>
              </Link>
            </h1>
          }
        </div>
      </div>
    </Dimmer>
    <style jsx>{`
      .menu-items{
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
      }

      a {
        color: inherit;
        cursor: pointer;
      }

      a:hover {
        color: #00ffff;
        cursor: pointer;
      }

      .fade {
        animation-duration: 500ms;
        animation-fill-mode: both;
      }

      @keyframes fadeIn {
        0% {opacity: 0;}
        100% {opacity: 0.8;}
      }

      .fadeIn {
        animation-name: fadeIn;
      }
    `}</style>
  </div>
)))
