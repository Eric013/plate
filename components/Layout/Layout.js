import Head from 'next/head'
import Navigation from '../Navigation/Navigation'

export default ({ children, menuOpen, openMenu, closeMenu }) => (
  <div>
    <style jsx global>{`
      * {
        margin: 0;
        box-sizing: border-box;
      }
      body {
        background-color: white;
      }
    `}</style>
    <Head>
      <meta charset='utf-8' />
      <meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no' />
      <meta name='theme-color' content='#00B5AD' />
      <meta name='apple-mobile-web-app-status-bar-style' content='black-translucent' />
      <link rel='shortcut icon' href='/static/plateLogo.png' />
      <link rel='stylesheet' href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.7/semantic.css' />
    </Head>
    <Navigation
      menuOpen={menuOpen}
      openMenu={openMenu}
      closeMenu={closeMenu}
    />
    {children}
  </div>
)
