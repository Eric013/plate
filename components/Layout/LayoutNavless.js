import Head from 'next/head'

export default ({ children }) => (
  <div>
    <style jsx global>{`
      * {
        margin: 0;
        box-sizing: border-box;
      }
      body {
        background-image: url('/static/plateGradient.png');
        font-family: 'Fira Sans', sans-serif;
      }
    `}</style>
    <Head>
      <meta charSet='utf-8' />
      <meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no' />
      <meta name='theme-color' content='#00B5AD' />
      <meta name='apple-mobile-web-app-status-bar-style' content='black-translucent' />
      <link rel='shortcut icon' href='/static/plateLogo.png' />
      <link href='https://fonts.googleapis.com/css?family=Fira+Sans' rel='stylesheet' />
      <link href='/static/flexboxgrid.css' rel='stylesheet' />
      <link rel='manifest' href='/static/manifest.json' />
    </Head>
    {children}
  </div>
)
