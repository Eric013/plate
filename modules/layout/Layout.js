import { Menu, Sidebar } from 'semantic-ui-react'
import React, { Component } from 'react'

import AuthenticatedMenu from './components/AuthenticatedMenu'
import Head from 'next/head'
import Navigation from '../navigation/Navigation'
import UnauthenticatedMenu from './components/UnauthenticatedMenu'
import { observer } from 'mobx-react'
import store from './store/store'

const MainLayout = observer(class MainLayout extends Component {
  openSidebar = () => {
    store.sidebarOpen = true
  }

  closeSidebar = () => {
    store.sidebarOpen = false
  }

  render() {
    return (
      <div>
        <style jsx global>{`
          html {
            margin: 0;
            padding: 5px;
            padding-top: 30px;
            font-family: sans-serif;
            background-color: #EFEFEF;
          }
        `}</style>
        <Head>
          <meta charset="utf-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no" />
          <meta name="theme-color" content="#00B5AD" />
          <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
          <link rel="stylesheet" href="https//cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.6/semantic.min.css"></link>
          <link rel="shortcut icon" href="/static/plateLogo.png" />
        </Head>
        <Sidebar
          as={Menu}
          animation='overlay'
          width='thin'
          visible={store.sidebarOpen}
          icon='labeled'
          color='teal'
          vertical
          inverted
        >
          {store.loggedIn
            ? <AuthenticatedMenu toggleSidebar={this.closeSidebar} />
            : <UnauthenticatedMenu toggleSidebar={this.closeSidebar} />
          }
        </Sidebar>
        <Sidebar.Pusher>
          <div>
            <Navigation toggleSidebar={this.openSidebar} />
            {this.props.children}
          </div>
        </Sidebar.Pusher>
      </div>
    )
  }
})

export default MainLayout
