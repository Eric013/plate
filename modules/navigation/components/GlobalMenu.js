import { inject, observer } from 'mobx-react'

import { Dimmer } from 'semantic-ui-react'
import Link from 'next/prefetch'

export default inject('store')(observer(({ store }) => (
  <div>
    <Dimmer active={store.menuOpen}
      onClickOutside={store.closeMenu}
      page
    >
      <div className='menu-items'>
        <h1 onClick={store.closeMenu}>
          <Link href='/'>
            <a>
              Home
            </a>
          </Link>
        </h1>
        <h1 onClick={store.closeMenu}>
          <Link href='/login'>
            <a>
              Login
            </a>
          </Link>
        </h1>
        <h1 onClick={store.closeMenu}>
          <Link href='/dashboard'>
            <a>
              Dashboard
            </a>
          </Link>
        </h1>
      </div>
    </Dimmer>
    <style jsx>{`
      .menu-items{
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
      }

      a {
        color: inherit;
      }
    `}</style>
  </div>
)))
