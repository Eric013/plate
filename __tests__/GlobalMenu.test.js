/* global it, expect, describe */

import GlobalMenu from '../components/GlobalMenu/GlobalMenu'
import Layout from '../components/Layout/Layout'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(
    <Layout>
      <GlobalMenu />
    </Layout>,
    div)
})

describe('GlobalMenu snapshot', () => {
  it('renders without crashing!"', () => {
    const component = renderer.create(
      <Layout>
        <GlobalMenu />
      </Layout>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
