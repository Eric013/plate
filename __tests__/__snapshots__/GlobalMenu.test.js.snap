exports[`GlobalMenu snapshot renders without crashing!" 1`] = `
<div
  data-jsx={4290012279}>
  <div
    className="nav-bar"
    data-jsx={2548831492}>
    <div
      className="ui pointing top fixed menu">
      <a
        data-jsx={2548831492}
        href="/"
        onClick={[Function]}>
        <div
          className="link item"
          onClick={[Function]}>
          <img
            data-jsx={2548831492}
            src="/static/plateLogo.png" />
          <span
            className="logo-text"
            data-jsx={2548831492}>
            late
          </span>
        </div>
      </a>
      <div
        className="right menu">
        <div
          className="link item"
          onClick={[Function]}>
          <span
            data-jsx={2548831492}>
            MENU
          </span>
        </div>
      </div>
    </div>
    <div
      data-jsx={2779333434} />
  </div>
  <div
    data-jsx={2779333434} />
</div>
`;
